packages/back-end/src/models/EventWebhookModel.ts

```diff
@@ -167,17 +167,33 @@ type EventWebHookDocument = mongoose.Document & EventWebHookInterface;
  * @param doc
  * @returns
  */
-const toInterface = (doc: EventWebHookDocument): EventWebHookInterface => {
+const toInterface = async (
+  doc: EventWebHookDocument
+): Promise<EventWebHookInterface> => {
   const payload = omit(doc.toJSON<EventWebHookDocument>(), ["__v", "_id"]);
 
+  // Add defaults values
+  const defaults = {
+    ...(payload.method ? {} : { method: "POST" }),
+    // All webhook are created with a payloadType. This is here for antiquated ones
+    // which don't have one and should be considered raw.
+    ...(payload.payloadType ? {} : { payloadType: "raw" }),
+    ...(payload.headers ? {} : { headers: {} }),
+    ...(payload.tags ? {} : { tags: [] }),
+    ...(payload.environments ? {} : { environments: [] }),
+  };
+
+  if (Object.keys(defaults).length)
+    await EventWebHookModel.updateOne(
```

Comment:
Can we do this without an `await`? We can let the update be handled in the background so this doesn't need to become an async function. Might also be related to the timeouts in CI

Reply:
Excellent suggestion, thanks!